lib/stripe.js

```diff
@@ -251,6 +261,50 @@ Stripe.prototype = {
       this[utils.pascalToCamelCase(name)] = new resources[name](this);
     }
   },
+
+  _getPropsFromConfig(config) {
+    // config can be an object or a string
+    const isString = typeof config === 'string';
+    const isObject = config === Object(config) && !Array.isArray(config);
+
+    if (!isObject && !isString) {
+      throw new Error('Config must either be an object or a string');
+    }
+
+    // if an api version is provided, first validate that
+    const version = isString ? config : config.apiVersion;
+    if (version && !utils.validateApiVersion(version)) {
+      throw new Error(
+        'Api version string should be in the format YYYY-MM-DD or "latest"'
+      );
+    }
+
+    let props = {};
+
+    // If config is a string, we assume the old behavior of passing in a string representation of the api version
+    if (isString) {
+      props.apiVersion = config;
```

Comment:
nit, I think it might make sense to early-return `{apiVersion: config}` here, since it'll be impossible to set other properties later anyway. 

Then, later on, we can just `return config`